src/main/java/io/jenkins/infra/repository_permissions_updater/hosting/Hoster.java

```diff
@@ -106,7 +112,10 @@ private void run(int issueID) {
 
             String prUrl = createUploadPermissionPR(issueID, forkTo, users, hostingRequest.getJenkinsProjectUsers(), issueTrackerChoice == IssueTracker.GITHUB, componentId);
             if (StringUtils.isBlank(prUrl)) {
-                LOGGER.error("Could not create upload permission pull request");
+                String msg = "Could not create upload permission pull request";
+                LOGGER.error(msg);
+                reportHostingFailure(issueID, msg);
+                return;
```

Comment:
Here it was not returning, so it would send the success message even when the creation of the uploadpermission PR failed.